app/api/api_v1/endpoints/report.py

```diff
@@ -0,0 +1,143 @@
+import datetime
+import json
+
+from fastapi import APIRouter, Depends, Response, HTTPException
+from fpdf import FPDF
+from sqlalchemy.orm import Session
+
+import crud
+import utils
+from api import deps
+from models import User
+from schemas import FarmProfile, MachineryAssetsOfFarm, PlotParcelDetail, ReportDB, ReportCreate, Message, ReportDBID
+
+from utils import plant_protection, irrigations, fertilisation, harvests, work_book
+
+router = APIRouter()
+
+
+@router.get("/{report_id}")
+def get_by_id(
+        report_id: int,
+        current_user: User = Depends(deps.get_current_user),
+        db: Session = Depends(deps.get_db)
+):
+    """
+    Returns a report via ID.
+    """
+
+    report_db = crud.report.get(db=db, id=report_id)
+
+    if not report_db:
+        raise HTTPException(
+            status_code=400,
+            detail="Report with ID:{} does not exist.".format(report_id)
+        )
+
+    types = {"work_book": "work-book", "plant_protection": "plant-protection", "irrigation": "irrigations", "fertilisations": "fertilisations", "harvest": "harvests", "global_gap": "GlobalGAP"}
+
+    if report_db.type == "work-book":
+        json_file = json.loads(report_db.file)
+
+        pdf = utils.work_book(
+            farm=utils.parse_farm_profile(json_file),
+            plot=utils.parse_plot_detail(json_file),
+            cult=utils.parse_generic_cultivation_info(json_file),
+            irri=utils.parse_irrigation(json_file),
+            fert=utils.parse_fertilization(json_file),
+            pdmd=utils.parse_plant_protection(json_file)
+        )
+
+    elif report_db.type == "plant-protection":
+
+        pdf = plant_protection(utils.parse_plant_protection(json.loads(report_db.file)))
```

Comment:
better to use it as `utils.plant_protection(...`

Reply:
Agreed, updated the usage for the utils functions